stock_picking_actual_date/models/stock_move.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2023-2024 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import pytz
+
+from odoo import api, fields, models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    actual_date = fields.Date(
+        compute="_compute_actual_date",
+        store=True,
+    )
+
+    @api.depends("date", "picking_id.actual_date")
+    def _compute_actual_date(self):
+        for rec in self:
+            if rec.picking_id.actual_date:
+                rec.actual_date = rec.picking_id.actual_date
+                continue
+            company_tz = self.env.company.tz or "UTC"
+            tz = pytz.timezone(company_tz)
+            # Convert the UTC date directly to the company's timezone
+            rec.actual_date = rec.date.replace(tzinfo=pytz.utc).astimezone(tz)
```

Comment:
Use `context_today()` instead, passing the company timezone (self.env.company.partner_id.tz) as the `tz` context if available, else self.env.user.tz.